spec/lib/roo/base_spec.rb

```diff
@@ -158,6 +158,11 @@ def sheets
         expect { spreadsheet.row_with([/Missing Header/]) }.to \
           raise_error(Roo::HeaderRowNotFoundError)
       end
+
+      it 'returns missing headers' do
+        expect { spreadsheet.row_with([/Header/, /Missing Header 1/, /Missing Header 2/]) }.to \
+          raise_error(Roo::HeaderRowNotFoundError, /(?=.*Missing Header 1)(?=.*Missing Header 2)/)
```

Comment:
This test will pass even if I raise `query` instead of `missing_headers`

Reply:
Totally true, decided to check for returned string, improves readability imho